### PR TITLE
Check crc on tar header

### PR DESCRIPTION
**Pull request by Erior**
#837

We could use the CRC for checking validation of the header, however it would seem the Tar.Empty.tar test case is invalid

### CODE REVIEW

File: tests/SharpCompress.Test/Tar/TarArchiveTests.cs [hunk 1]
```diff
@@ -217,8 +217,7 @@ public void Tar_Empty_Archive()
     {
         var archiveFullPath = Path.Combine(TEST_ARCHIVES_PATH, "Tar.Empty.tar");
         using Stream stream = File.OpenRead(archiveFullPath);
-        using var archive = ArchiveFactory.Open(stream);
-        Assert.True(archive.Type == ArchiveType.Tar);
+        Assert.Throws<InvalidOperationException>(() => ArchiveFactory.Open(stream));
```
Comment (adamhathcock) [inline tests/SharpCompress.Test/Tar/TarArchiveTests.cs]: Feels like not being able to open an empty tar is a regression

**Review by adamhathcock — CHANGES_REQUESTED** [review]

File: tests/SharpCompress.Test/Tar/TarArchiveTests.cs (hunk 1 above)
Comment (Erior) [reply tests/SharpCompress.Test/Tar/TarArchiveTests.cs]: You are correct, the empty file is a special case that needs to be taken care of, updated solution proposal

**Comment by Erior** [comment]
The last could actually be change to return crc == chk1 || crc == chk2   ... instead of the if... if you prefere that I can change

**Review by adamhathcock — APPROVED** [review]

**Comment by adamhathcock** [comment]
Thanks!

**Pull request merged**